CardinalKit-Example/CardinalKit/Library/HealthKit/CKHealthKitManager.swift

```diff
@@ -56,4 +237,16 @@ class CKHealthKitManager : NSObject {
         }
     }
     
+    
+    func collectAllTypes(_ completion: @escaping (_ success: Bool, _ error: Error?) -> Void) {             
+        // handle authorization from the OS
+        CKActivityManager.shared.getHealthAuthorizaton(forTypes: hkTypesToReadInBackground) {(success, error) in
```

Comment:
Love this function!

Let's jump on a call to discuss the functionality of:
`CKActivityManager` :: `startHealthKitCollectionInBackground()`

There are some considerations that could lead to bugs -- CardinalKit does not support the [HK background delivery API ](https://developer.apple.com/documentation/healthkit/hkhealthstore/1614175-enablebackgrounddelivery) for all HKSampleTypes yet.